code/game/objects/structures/window_frames.dm

```diff
@@ -54,17 +54,47 @@
 /obj/structure/window_frame/Initialize(mapload)
 	. = ..()
 
+	if(grille)
+		grille = new grille(src, sheet_amount)
+
 	update_appearance()
 	AddComponent(/datum/component/climb_walkable)
 	AddElement(/datum/element/climbable, on_try_climb_procpath = TYPE_PROC_REF(/obj/structure/window_frame, on_try_climb))
 
+	var/static/list/tool_tips = list(
+		TOOL_WELDING = list(
+			SCREENTIP_CONTEXT_LMB = "Repair",
+			SCREENTIP_CONTEXT_RMB = "Deconstruct",
+		),
+		TOOL_WIRECUTTER = list(
+			SCREENTIP_CONTEXT_LMB = "Remove grille",
+		),
+
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_tips)
```

Comment:
Screentips on both the Tool-side and for the iron rod/grill sides of things are not showing up in testing. Glass however is showing up correctly.